src/backend/cdb/motion/tupleremap.c

```diff
@@ -268,7 +268,7 @@ TRRemapDatum(TupleRemapper *remapper, Oid typeid, Datum value)
 
```

Comment:
I suppose an extra `if (value == 0)` should be added before the call to `BuildTupleRemapInfo()`. In the case in question, we do not need the result of the method, and it does not have any side-effects (except for draining space from `remapper->mycontext`).

Reply:
Good catch! Thanks, fixed.